### PR TITLE
Short circuited clipGeometry for Points to improve performance

### PR DESCRIPTION
**Pull request by timrobertson100**
This is discussed in https://github.com/ElectronicChartCentre/java-vector-tile/issues/13

Benchmarking this shows around 30 times improvement for the "test harness" included.

(Please accept my apologies for reformatting where double lines and unnecessary spaces were also cleaned up - a sloppy commit).

### CODE REVIEW

File: src/main/java/no/ecc/vectortile/VectorTileEncoder.java [hunk 1]
```diff
@@ -169,11 +174,23 @@ public void addFeature(String layerName, Map<String, ?> attributes, Geometry geo
 
         layer.features.add(feature);
     }
-    
+
+    /**
+     * A short circuit clip to the tile extent (tile boundary + buffer) for points to improve performance.
+     * @see https://github.com/ElectronicChartCentre/java-vector-tile/issues/13
+     */
+    protected Geometry clipToExtent(Point geom) {
```
Comment (halset) [inline src/main/java/no/ecc/vectortile/VectorTileEncoder.java]: If this method is not to change the Point, then perhaps returning a boolean would be better? That way, you do not need to return a empty geometry?

Comment (timrobertson100) [inline src/main/java/no/ecc/vectortile/VectorTileEncoder.java]: Yes.

If you are in agreement, perhaps you could cherry pick from this PR to do a short circuited check if addFeature should continue for points?  I'm happy to do another PR, but it is really just 4 lines of code.

Thank you for paying attention to feedback - it's greatly appreciated.

**Comment by halset** [comment]
Looks very good. I will do a small refactoring and commit.

**Comment by timrobertson100** [comment]
Thanks!

**Comment by halset** [comment]
released 1.1.1 with this fix

**Comment by timrobertson100** [comment]
Thanks again
